collections/_posts/2023-04-03-typelevel_toolkit.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title: Typelevel Toolkit
+category: technical
+
+meta:
+  nav: blog
+  author: zetashift
+  pygments: true
+---
+
+Getting started in the wondrous world of functional programming using [Typelevel libraries](https://typelevel.org/projects) can be daunting. Before you can even write your first pure "Hello, World!" you'll need to install a Java runtime, editor tooling and build tools. Then you'll need to setup some project using [sbt]() or [mill](). As an added consequence, after all the setup, the idea of using these battle-tested libraries for small scripts will seem like a chore. This is where [Typelevel Toolkit](https://typelevel.org/toolkit/) comes in. It provides an easy start for beginning and experienced developers with Scala and functional programming.
+
+# `scala-cli` to the rescue
+
+[scala-cli](https://scala-cli.virtuslab.org/) is a command-line interface to quickly develop and experiment with Scala, it's even on track to [becoming the new scala command](https://docs.scala-lang.org/sips/scala-cli.html). The interface has a lot of advantages, but one of the most important ones is that it makes learning, developing and building Scala scripts and small applications friction-less and easy to use.
+
+You can get `scala-cli` by following the installation instructions described here: [https://scala-cli.virtuslab.org/docs/overview](https://scala-cli.virtuslab.org/docs/overview#installation)#installation. The great part here is that once you have `scala-cli` installed, it will take care of the rest: Java runtimes, editor tooling, compilation targets(including a native target!) and you can even use dependencies in your scripts!
+
+## An example of setting up your script
+
+First off, let's create a new directory that will contain our script(s).
+
+```sh
+mkdir myscript && cd myscript
+```
+
+Now we can use `scala-cli` to create all the files necessary for editor tooling to work:
+
+```sh
+scala-cli setup-ide .
+```
+
+Finally, let's create a Scala file and try to compile it.
+
+```sh
+touch Main.scala
+scala-cli compile --watch Main.scala
+```
+
+
+The last command also includes a `--watch` flag, so we can hack on our script and `scala-cli` will try to compile the file on every save.
+This creates a very nice feedback loop!
+
+# Putting the fun in functional
+
+[Typelevel Toolkit](https://typelevel.org/toolkit/) uses `scala-cli` and Typelevel libraries to provide a runway for your next Scala script or command-line interface. You can use the toolkit with `scala-cli` with just a single line and you'll get [Cats Effect](https://typelevel.org/cats-effect/), [fs2](https://fs2.io) and a few other libraries to develop scripts quickly using pure functional programming. 
```

Comment:
Maybe give the complete list of libraries somewhere in the post? I think many readers will be curious exactly what is included.

Reply:
So here I give some more hints to what libraries are available https://github.com/typelevel/typelevel.github.com/pull/439/files#diff-62195842897f509fcd7512ee548637afed3f916d23416d51358590c220a595e9R82

And here I link to the full overview: https://github.com/typelevel/typelevel.github.com/pull/439/files#diff-62195842897f509fcd7512ee548637afed3f916d23416d51358590c220a595e9R75 

I didn't want to completely list out the libraries, because they might change over time.

How about I put that overview link on the "a few other libraries" text?

Reply:
> I didn't want to completely list out the libraries, because they might change over time.

I think it's fine :) a blog-post isn't supposed to be ever-lasting documentation, it's a dated post.

Reply:
LGTM

Reply:
I've changed up the paragraphs that list libraries included in the toolkit by quite a lot. Let me know how y'all like it.

Reply:
LGTM